antarest/login/web.py

```diff
@@ -41,13 +40,31 @@ def login() -> Any:
 
         # Identity can be any data that is json serializable
         access_token = create_access_token(identity=user.to_dict())
-        resp = jsonify({"user": user.name, "access_token": access_token})
-        # set_access_cookies(resp, access_token)
+        refresh_token = create_refresh_token(identity=user.to_dict())
+        resp = jsonify(
+            {
+                "user": user.name,
+                "access_token": access_token,
+                "refresh_token": refresh_token,
+            }
+        )
+
         return (
             resp,
             200,
         )
 
+    @bp.route("/refresh", methods=["POST"])
+    @auth.protected()
+    def refresh() -> Any:
+        identity = get_jwt_identity()
+        user = service.get_user(identity["id"])
+        if user:
+            access_token = create_access_token(identity=identity)
+            return jsonify(access_token=access_token)
+        else:
+            return "Token invalid", 403
```

Comment:
Not sure if this is right (https://flask-jwt-extended.readthedocs.io/en/stable/refreshing_tokens/#explicit-refreshing-with-refresh-tokens)

Reply:
I really don't understand flask-jwt example. If server just create same access_token with new expiration date. Why we just don't expend token expiration date ? For me we use a short-time access_token with a long-time refresh token, to force user to sync frequently information in his access_token.

Each time user ask new access_token, we have to update its new groups,  permissions or blacklisted.

However, in the example, they use `@jwt_required(refresh=True)`, it's a better way than `@auth.protected()`. I  tried to implements it, but I can't find how to correctly send refresh_token during test.  We can come back to this issue later.

Reply:
In the example you have to replace the token in headers ('Bearer <token>') with the refresh_token instead of the access_token (it is documented at the last line of the linked section).

Reply:
yes, but it doesn't work... only for test. It test with curl, refresh endpoint work great, but with test it failed